### PR TITLE
Clarify what happens when setting the audio volume

### PR DESCRIPTION
**Pull request by hate**
# Objective

- Fixes [#8835](https://github.com/bevyengine/bevy/issues/8835)

## Solution

- Added a note to the `set_volume` docstring which explains how volume is interpreted.

### CODE REVIEW

File: crates/bevy_audio/src/sinks.rs [hunk 1]
```diff
@@ -15,6 +15,11 @@ pub trait AudioSinkPlayback {
     ///
     /// The value `1.0` is the "normal" volume (unfiltered input). Any value other than `1.0`
     /// will multiply each sample by this value.
+    ///
+    /// Note on Audio Volume:
```
Comment (alice-i-cecile) [inline crates/bevy_audio/src/sinks.rs:19]:
```suggestion
    /// # Note on Audio Volume
```

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
I think that the suggestion will give nicer formatting, but otherwise this LGTM.

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_audio/src/sinks.rs [hunk 2]
```diff
@@ -15,6 +15,11 @@ pub trait AudioSinkPlayback {
     ///
     /// The value `1.0` is the "normal" volume (unfiltered input). Any value other than `1.0`
     /// will multiply each sample by this value.
+    ///
+    /// # Note on Audio Volume
```
Comment (mockersf) [inline crates/bevy_audio/src/sinks.rs:19]:
```suggestion
    /// # Note on Audio Volume
    ///
```
not a hard rule, but it's nicer to keep a new lines after heading

**Comment by mockersf** [comment]
I think I don't agree with the comment. The video is audio at 1.0, then 0.01, then 1.0. for me it's much less than half the volume.

https://github.com/bevyengine/bevy/assets/8672791/cc144fcd-c728-428e-8c53-749b207af414

**Comment by alice-i-cecile** [comment]
That would be a quarter, based on the comment, but that's still perceived as much less than that to me.

**Comment by GitGhillie** [comment]
Sound is doubled with every 3dB, so to get half the sound you would need -3dB, or a power of 10^-3/10 ~= 0.5. If we take the square root of 0.5 you get 0.708 which is what you would need to set the volume to in order to halve the sound from the original 1.0 setting. Multiply the volume by 0.708 to halve the sound again etc. See table below:

image

To make things more complicated:
"an increase of 10 dB roughly corresponds to the perceived volume doubling in intensity"

This means you would have to set the volume to 0.316 to perceive it as twice as quiet as 1.0.
And 0.316 * 0.316 to perceive it 4 times as quiet as 1.0.

These numbers seem to check out when I try it on the example, as well as the change in dB levels when I check with OBS.

**Review by harudagondi — APPROVED** [review]
I agree with the documentation, but I am wondering if we should have it accept something like
```rust
enum Volume {
    Logarithmic(f32),
    Linear(u8), // 255 is the 100%, 128 is 50% of the volume, etc.
}
```
instead of just `f32`.

File: crates/bevy_audio/src/sinks.rs [hunk 3]
```diff
@@ -15,6 +15,11 @@ pub trait AudioSinkPlayback {
...
+    /// Our perception of audio volume is based on a logarithmic scale. As such, setting the volume to 0.01
+    /// doesn't mean it's 100 times quieter. Instead, it's exactly two times quieter than the default.
+    /// Mathematically, if 10^x = 0.01, then x = -2, representing a factor of two.
```
Comment (GitGhillie) [inline crates/bevy_audio/src/sinks.rs:22]:
```suggestion
    /// An increase of 10 decibels (dB) roughly corresponds to the perceived volume doubling in intensity.
    /// As this function scales not the volume but the amplitude, a conversion might be necessary.
    /// For example, to halve the perceived volume you need to decrease the volume by 10 dB.
    /// This corresponds to 20log(x) = -10dB, solving x = 10^(-10/20) = 0.316.
    /// Multiply the current volume by 0.316 to halve the perceived volume.
```
Technically `set_volume` is misleading as it scales the amplitude, not the volume. Usually audio volume is in dB by definition.

IMO it should be called something like `set_amplitude` and then we can have function like `set_volume_percentage(f32)` where we do the conversion (which will be subjective). But for this PR let's just make sure the comment is correct and helpful. 

@harudagondi have you tried setting the volume to 0.01? What we're saying is that setting it to 0.01 makes the perceived volume wayyy less than two times quieter.

**Review by GitGhillie — CHANGES_REQUESTED** [review]

File: crates/bevy_audio/src/sinks.rs (hunk 3 above)
Comment (harudagondi) [reply crates/bevy_audio/src/sinks.rs:22]: Hmm that's true actually. Let's just merge this PR and then add a deprecate warning on `set_volume`, or possibly just have the enum Volume thing like i suggested.

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**